helm-chart/PF_RING.md

```diff
@@ -0,0 +1,189 @@
+            - [Steps to Use kmm with Custom Containers](#steps-to-use-kmm-with-custom-containers)
+    - [Appendix A: PF_RING kernel module compilation](#appendix-a-pf_ring-kernel-module-compilation)
+        - [Automated complilation](#automated-complilation)
+        - [Manual compilation](#manual-compilation)
+
+<!-- /TOC -->
+
+## Overview
+
+PF_RING™ is an advanced Linux kernel module and user-space framework designed for high-speed packet processing. It offers a uniform API for packet processing applications, enabling efficient handling of large volumes of network data.
+
+For comprehensive information on PF_RING™, please visit the [User's Guide]((https://www.ntop.org/guides/pf_ring) and access detailed [API Documentation](http://www.ntop.org/guides/pf_ring_api/files.html).
+
+## Provisioning mode
+
+There are two approaches for loading the PF_RING kernel module on nodes:
+
+1. `auto`
+
+In this mode, the Kubeshark worker retrieves the necessary PF_RING kernel module version from an S3 bucket and loads it onto the node.
+
+> mode=auto requires an active internet connection and is not suitable for air-gapped environments.
+
+2. `kmm`
+
+The Kernel Module Management controller ([KMM](https://kmm.sigs.k8s.io/documentation/deploy_kmod/)) acquires the required PF_RING kernel module version from a Docker container and loads it onto the node
+
+> mode=kmm is suitable for air-gapped environments. 
+
+## Selection of Provisioning Mode
+
+> This step is optional. In case mode=auto and no PF_RING kernel module is found Kubeshark falls back to `libpcap` if `PF_RING` kernel module not available 
+
+Prior to choosing a method, it is essential to verify if a PF_RING kernel module is already built for your kernel version.
+Kubeshark provides additional CLI tool for this purpose - [pf-ring-compiler](https://github.com/kubeshark/pf-ring-compiler).
+
+Compatibility verification can be done by running:
+
+```
+pfring-compiler compatibility
+```
+
+This command checks for the availability of kernel modules for the kernel versions running across all nodes in the Kubernetes cluster.
+
+Example output for a compatible cluster:
+```
+Node                                          Kernel Version                 Supported
+ip-192-168-77-230.us-west-2.compute.internal  5.10.199-190.747.amzn2.x86_64  true
+ip-192-168-34-216.us-west-2.compute.internal  5.10.199-190.747.amzn2.x86_64  true
+
+Cluster is compatible
+```
+
+
+### Pre-built kernel module exists and external egress allowed
+
+If PF_RING kernel modules are already available for the target nodes (cluster is compatible), both `auto` and `kmm` modes are applicable.
+
+|auto|kmm|
+|----|---|
+| `SYS_MODULE` capability required for Kubeshark | `SYS_MODULE` capability is **not** required for Kubeshark|
+| no additional dependencies | (!)requires `cert-manager` and `KMM` installed (follow [instructions](https://kmm.sigs.k8s.io/documentation/install/)) |
```

Comment:
@corest 
When trying to install KMM on EKS I get the pd pending. 
I use this:
```yaml
kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.11.0/cert-manager.yaml
kubectl -n cert-manager wait --for=condition=Available deployment \
    cert-manager \
    cert-manager-cainjector \
    cert-manager-webhook
kubectl apply -k https://github.com/kubernetes-sigs/kernel-module-management/config/default
```
Only the last command doesn't finish. 
<img width="766" alt="image" src="https://github.com/kubeshark/kubeshark/assets/1990761/12dc6a25-62a1-499d-a344-27f7fd446382">
I get the following warning:
```yaml
│   ----     ------            ----                   ----               -------                                                                                                                             │
│   Warning  FailedScheduling  3m41s (x2 over 8m52s)  default-scheduler  0/1 nodes are available: 1 node(s) didn't match Pod's node affinity/selector. preemption: 0/1 nodes are available: 1 Preemption is  │
│ not helpful for scheduling..     
```

Reply:
@alongir yeah, kmm runs on control-plane, but there is no access to control-plane. So you need to label one of workers as control-plane. This is why I had discussion about this in kubernetes slack https://kubernetes.slack.com/archives/C037RE58RED/p1702334359834029 and created the issue https://github.com/kubernetes-sigs/kernel-module-management/issues/666